code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -41,24 +43,30 @@
 	if(in_range(user, src) || isobserver(user))
 		. += "<span class='notice'>The status display reads: This unit can hold a maximum of <b>[max_n_of_items]</b> items.</span>"
 
-/obj/machinery/smartfridge/update_icon()
-	if(!machine_stat)
-		if (visible_contents)
-			switch(contents.len)
-				if(0)
-					icon_state = "[initial(icon_state)]"
-				if(1 to 25)
-					icon_state = "[initial(icon_state)]1"
-				if(26 to 75)
-					icon_state = "[initial(icon_state)]2"
-				if(76 to INFINITY)
-					icon_state = "[initial(icon_state)]3"
-		else
-			icon_state = "[initial(icon_state)]"
-	else
+/obj/machinery/smartfridge/update_icon_state()
+	if(machine_stat)
 		icon_state = "[initial(icon_state)]-off"
+		return ..()
 
+	if(!visible_contents)
+		icon_state = "[initial(icon_state)]"
+		return ..()
 
+	switch(contents.len)
+		if(0)
+			icon_state = "[initial(icon_state)]"
+		if(1 to 25)
+			icon_state = "[initial(icon_state)]1"
+		if(26 to 75)
+			icon_state = "[initial(icon_state)]2"
+		if(76 to INFINITY)
+			icon_state = "[initial(icon_state)]3"
+	return ..()
+
+/obj/machinery/smartfridge/update_overlays()
+	. = ..()
+	if(!machine_stat && has_emissive)
+		. += emissive_appearance(icon, "[initial(icon_state)]-light-mask", alpha = src.alpha)
```

Comment:
This also needs to update lum sources to get the proper luminosity set

Reply:
makes sense, this was the only one I was having trouble with getting the emissive to show up.

Reply:
addressed in 1389b29